strongbox-web-core/src/main/java/org/carlspring/strongbox/rest/ArtifactController.java

```diff
@@ -0,0 +1,517 @@
+import org.carlspring.strongbox.storage.ArtifactResolutionException;
+import org.carlspring.strongbox.storage.ArtifactStorageException;
+import org.carlspring.strongbox.storage.Storage;
+import org.carlspring.strongbox.storage.metadata.MetadataType;
+import org.carlspring.strongbox.storage.repository.Repository;
+import org.carlspring.strongbox.util.MessageDigestUtils;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import javax.xml.bind.JAXBException;
+import java.io.*;
+import java.net.URLEncoder;
+import java.nio.file.Paths;
+import java.security.NoSuchAlgorithmException;
+import java.text.SimpleDateFormat;
+import java.util.Arrays;
+import java.util.Date;
+import java.util.Locale;
+import java.util.regex.Matcher;
+
+import com.google.common.io.ByteStreams;
+import io.swagger.annotations.ApiOperation;
+import io.swagger.annotations.ApiResponse;
+import io.swagger.annotations.ApiResponses;
+import org.apache.commons.io.FileUtils;
+import org.apache.commons.io.comparator.DirectoryFileComparator;
+import org.apache.log4j.LogManager;
+import org.apache.log4j.Logger;
+import org.apache.maven.artifact.repository.metadata.Metadata;
+import org.codehaus.plexus.util.xml.pull.XmlPullParserException;
+import org.springframework.http.*;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.web.bind.annotation.*;
+import static org.carlspring.strongbox.rest.ByteRangeRequestHandler.handlePartialDownload;
+import static org.carlspring.strongbox.rest.ByteRangeRequestHandler.isRangedRequest;
+import static org.springframework.http.HttpStatus.INTERNAL_SERVER_ERROR;
+import static org.springframework.http.HttpStatus.NOT_FOUND;
+
+@RestController
+@RequestMapping("/storages")
+public class ArtifactController
+        extends BaseArtifactController
+{
+
+    private static final Logger logger = LogManager.getLogger(ArtifactController.class.getName());
+
+
+    @PreAuthorize("authenticated")
+    @RequestMapping(value = "greet", method = RequestMethod.GET)
+    public ResponseEntity greet()
+    {
+        return new ResponseEntity<>("success", HttpStatus.OK);
+    }
+
+    @ApiOperation(value = "Used to deploy an artifact", position = 0)
+    @ApiResponses(value = { @ApiResponse(code = 200, message = "The artifact was deployed successfully."),
+                            @ApiResponse(code = 400, message = "An error occurred.") })
+    @PreAuthorize("hasAuthority('ARTIFACTS_DEPLOY')")
+    @RequestMapping(value = "{storageId}/{repositoryId}", method = RequestMethod.PUT)
+    public ResponseEntity upload(
+                                        @PathVariable String storageId,
+                                        @PathVariable String repositoryId,
+                                        @RequestParam(name = "path") String path,
+                                        HttpEntity<byte[]> requestEntity)
```

Comment:
@AlexOreshkevich ,

I don't think this is a good idea, as I believe it will map the entire byte array in memory which will, in term, make it crash with larger artifacts. I believe this used to work with an `InputStream` previously.